### PR TITLE
Fix for TypeScript test on Windows

### PR DESCRIPTION
**Pull request by tbroadley**
The tests `ReactTypeScriptClass` and `ReactClassEquivalence` fail on Windows because of file paths using backslashes. This PR fixes the problem.

### CODE REVIEW

File: scripts/jest/ts-preprocessor.js [hunk 1]
```diff
@@ -19,7 +19,16 @@ function compile(content, contentFilename) {
   var output = null;
   var compilerHost = {
     getSourceFile: function(filename, languageVersion) {
-      var source;
+      var source, reactRegex;
+
+      // Accomodations for backslashes in Windows file paths.
+      if (process.platform === 'win32') {
+        filename = filename.replace(/\//g, '\\');
```
Comment (zpao) [inline scripts/jest/ts-preprocessor.js]: Can we use `path` APIs to do this without having to do it manually? Eg `path.normalize` (no idea if it works, just that it exists). Might be able to swing that for the regexp too (eg `new RegExp(path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$'))`

**Comment by zpao** [comment]
PS, thanks for coming in and helping out with the Windows stuff! I don't know if you can tell but uhhh we don't have a lot of people doing that :)

**Comment by tbroadley** [comment]
Thanks for the feedback! I'll look at path.normalize. And yeah I figured that about Windows ;) I know that there's one bug with one of the other tests, but haven't had a chance to look at it it yet.

**Comment by zpao** [comment]
Awesome, thanks!

**Pull request merged**